pkg/cache/v3/cache.go

```diff
@@ -21,15 +21,16 @@ import (
 	"fmt"
 	"sync/atomic"
 
-	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"google.golang.org/protobuf/types/known/anypb"
 
 	"github.com/golang/protobuf/proto"
```

Comment:
If you are switching proto libraries, should you switch over fully everywhere?

Reply:
Ah oops I just missed that one thank you

Reply:
So for this specific case I wasn't able to switch over since the underlying proto files still use `github.com/golang/protobuf/proto`, and to do a message reflect you need to use the corresponding lib. I was at least able to move away from the deprecated func though

Reply:
Yeh, when you mix & match, you end up having to convert using [MessageV2](https://pkg.go.dev/github.com/golang/protobuf/proto#MessageV2). For the v1 release, I'd recommend making sure the public API is fully V1 or fully V2 :)

Reply:
Ah okay, i'll revist this then